extender/ScanMonitor.js

```diff
@@ -0,0 +1,88 @@
+// This script monitors the active scanner and ends the scan if certain conditions are met.
+// By default it just ends the scan for high:
+//  * Connection failures
+//  * Authentication failures
+//  * Response times
+// You can easily chane the script to end the scan for other conditions, such as high 4xx / 5xx response codes,
+// but these tend to be application specific so they are not enabled by default.
+
+var SessionStructure = Java.type("org.zaproxy.zap.model.SessionStructure");
+var Timer = Java.type("java.util.Timer");
+var TimerTask = Java.type("java.util.TimerTask");
+var URI = Java.type("org.apache.commons.httpclient.URI");
+
+var extAscan = control.getExtensionLoader().getExtension("ExtensionActiveScan");
+var inMemoryStats = control
+  .getExtensionLoader()
+  .getExtension("ExtensionStats")
+  .getInMemoryStats();
+
+var timer = new Timer();
+var timerSecs = 10 * 1000; // Check every 10 secs
+
+// Set to true to see the stats reported live
+var log = false;
+
+function install(helper) {
+  timer.scheduleAtFixedRate(
+    new TimerTask()
+    {
+      run() {
+        runchecks();
+      },
+    },
+    0,
+    timerSecs
+  );
+}
+
+function getStat(site, stat) {
+  var val =
+    site == null
+      ? inMemoryStats.getStat(stat)
+      : inMemoryStats.getStat(site, stat);
+  return val == null ? 0 : val;
+}
+
+// Response times are recorded in logarithmic time slices
+function getLongRespStats(site) {
+  return (
+    getStat(site, "stats.responseTime.16384") +
```

Comment:
Are these ex: 16,32,64 sec? If so worth commenting?

Reply:
`Response times are recorded in logarithmic time slices` above :)

Reply:
Clarify that it's ms then?

Reply:
Done